File: problem-007/main.py
import math


def is_prime(x):
    # Correct for x >= 1
    # Any number x can only have 1 prime factor greater than sqrt(x)
    # Therefore if there are no factors <= sqrt(x) then x itself must be prime
    for i in range(2, int(math.sqrt(x)) + 1):
        if x % i == 0:
            return False
    return True


n = 2
x = 3

while n < 10_001:
    x += 2
    if is_prime(x):
        n += 1
        print(f"{n:06}", end="\r")

print(x)

File: problem-007/main.cpp
#include <iostream>
#include <cmath>


using namespace std;


bool is_prime(int x)
{
	int r = int(sqrt(x));

	for (int i=2; i<=r; i++){
		if (x % i == 0){
			return false;
		}
	}

	return true;
}


int main()
{
	int n = 2;
	int x = 3;

	while (n < 10001)
	{
		x += 2;

		if (is_prime(x))
		{
			n++;
		}
	}

	cout << x;
}

